### ISSUE TITLE
Add a Code Climate JSON formatter compatible with GitLab CI

### ISSUE THREAD

**Issue by pismy** [hadolint/hadolint#521]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

In GitLab CI, use `hadolint` **Code Climate** formatter to benefit from [GitLab Code Climate integration](https://docs.gitlab.com/ee/user/project/merge_requests/code_quality.html).

### Actual behavior

This doesn't work for 2 reasons.

1. `hadolint` outputs issues in **streamed format** (each issue being separated with a null char) while GitLab is expecting a non-streamed format, standard JSON array representation (each issue part of an array `[]`, separated by with coma)
2. GitLab is expecting the optional `fingerprint` field to be valuated

See: [GitLab CI expected Code Climate report format](https://docs.gitlab.com/ee/user/project/merge_requests/code_quality.html#implementing-a-custom-tool).

### Steps to reproduce the behavior

In GitLab, create a project with a faulty `Dockerfile` and the following `.gitlab-ci.yml` file:

```yaml
stages:
  - lint

hadolint:
  image:
    name: "hadolint/hadolint:latest"
    entrypoint: [""]
  stage: lint
  script:
    - hadolint -f codeclimate Dockerfile > hadolint.json
  artifacts:
    when: always
    reports:
      codequality: hadolint.json
```

GitLab will complain about "failed to parse code quality report".

**Comment by pismy** [comment]
PR to come...
